Add simple keyboard input player movement example.

Ablexxive opened the pull request:
Hi, I'm working on a little game for myself and it would have helped me to have an example of how to move a player character that has a rigid body and collider. Now that I got it working, I figured I'd pass it on, if you think it would make a good addition :)

rmsc commented on bevy_rapier2d/examples/player_movement.rs:
```diff
@@ -0,0 +1,74 @@
+            sprite: Sprite::new(Vec2::new(sprite_size_x, sprite_size_y)),
+            ..Default::default()
+        })
+        .with(RigidBodyBuilder::new_dynamic().can_sleep(false))
```
You don't actually need the `.can_sleep(false)` here. The problem you're experiencing is likely because you're setting linvel directly instead of using `set_linvel()`

rmsc commented on bevy_rapier2d/examples/player_movement.rs:
```diff
@@ -0,0 +1,74 @@
+        // Write player rigid_body velocity directly to component,
+        // the bevy_rapier plugin will update the Sprite transform.
+        if let Some(mut rb) = rigid_bodies.get_mut(rigid_body_component.handle()) {
+            rb.linvel = move_delta * player.0;
```
Here you should do `rb.set_linvel(move_delta * player.0, true)`. The `true` wakes up the body.

Ablexxive replied:
Ah thank you! Updated.

Ablexxive commented:
Addressed comments and renamed file to match other example patterns.

Ablexxive commented:
Just wanted to check in, is there anything I can do to wrap up this PR? :)

Ablexxive commented on bevy_rapier2d/examples/player_movement2.rs:
```diff
@@ -0,0 +1,76 @@
+    let sprite_size_y = 40.0;
+
+    // Spawn entity with `Player` struct as a component for access in movement query.
+    // If `can_sleep` is not set to false, entity will not respond to input after idle time.
```
Ah need to remove this comment, no longer applies.

rmsc commented:
> Just wanted to check in, is there anything I can do to wrap up this PR? :)

Sorry but I'm not a dev on this project, I just happened to be attempting the same thing and found this example useful.

sebcrozet reviewed (changes requested):
Hi @Ablexxive, thank you for this PR and very sorry for the late review!
And thank you @rmsc for your initial review!

This looks good overall, except for one details: we don't want to encourage people to use `1 pixel = 1 physics length unit`. This is a common bad practice, and this can lead to various problems when the game becomes bigger (potential float rounding problems, gravity that looks too weak, etc.) Therefore:
- The player collider should not have a size equal to 40 (because that would mean your player is 40 meters tall). We should give keep its size smaller, for example, 2, and set the `RapierConfiguration::scale` to `20.0`. This will tell Rapier to multiply the rigid-body coordinates by 20 when it overwrites the sprite position.
- As a result the initial velocity of the player should by for example 15 (which is 300 / 20).
- Finally the `move_delta` should be divided by the physics scale: `move_delta /= move_delta.magnitude() * rapier_parameters.scale`.

Ablexxive commented:
@sebcrozet No worries at all, I'm happy to be able to contribute. And @rmsc thanks so much for the review too :)

I've updated the code to use `RapierConfiguration::scale`. I had no idea about this and I'm glad to learn it, so that you for explaining! Some notes for the example:

- I left the sprite size in pixels and then used the scale factor along with a comment explaining why.
- The `300.0` in the `Player` component is actually movement speed (in `pixels/second`) not an initial velocity, so I left it at `300.0` and added comments in `fn player_movement` explaining that the division by physics scale is to convert from `pixels` to `physics_units`.

I could have just set the movement speed to `15` like you suggested but this way the example includes getting/using `rapier_parameters` in the movement code. Happy to change it if you prefer :)

sebcrozet commented:
All good. Thank you!

Ablexxive commented:
Great thanks!

onx2 commented:
Noobie question about movement & this example — I’m trying to understand this example as well as [bevy’s example](https://github.com/bevyengine/bevy/blob/main/examples/movement/physics_in_fixed_timestep.rs) and want to know if I need to reconcile the two approaches or if this one considers frame rate and interpolating visuals?
